feat(ui): support running as menubar app

awe123343 opened the pull request:
Close https://github.com/GopeedLab/gopeed/issues/1153

This PR
- [x] Support running `gopeed` as menu bar app on `MacOS`
- [x] Support close window with `Cmd+w` same as other MacOS apps

<img width="1401" height="974" alt="Screenshot 2025-12-28 at 22 17 16" src="https://github.com/user-attachments/assets/094770e6-4d3c-4a5f-800e-fee9dbcaeb56" />
<img width="1445" height="973" alt="Screenshot 2025-12-28 at 22 17 51" src="https://github.com/user-attachments/assets/268efd58-a264-401c-8e48-0f35ec8cfd69" />

awe123343 commented:
> Thanks for the PR! I've reviewed the changes and have a few suggestions:
> 
> 1. The `windows_manager` plugin supports hiding the dock icon, so there is no need to write a custom native MethodChannel. You can do this as:
>    ```dart
>    await windowManager.setSkipTaskbar(true);
>    ```
> 2. I noticed that the `runAsMenubarApp` option is stored in two places. I suggest keeping only the implementation in `database.dart`.
> 3. For the shortcut implementation, I recommend using the `hotkey_manager` package.

Thanks for the review! I've addressed all your comments:
  1. Removed custom `MethodChannel` - Now using `windowManager.setSkipTaskbar(true)` from `window_manager` plugin instead of custom native code. The `dock_icon.dart` file and related Swift `MethodChannel` code have been removed.
  2. Single source of truth for `runAsMenubarApp` - The setting is now stored only in `database.dart`.
    Note: We also sync the value to SharedPreferences so that native Swift can read it in `applicationWillFinishLaunching` - this is necessary to hide the dock icon early in the app lifecycle before Flutter fully initialises, preventing a brief dock icon flash on startup.
  3. Using `hotkey_manager` for Cmd+W - Replaced the custom Swift performKeyEquivalent override with `hotkey_manager` package.

monkeyWie commented on ui/flutter/macos/Runner/AppDelegate.swift:
I think the modification of this file is unnecessary because the `main.dart` file already handles it.

monkeyWie commented on ui/flutter/lib/main.dart:
```diff
@@ -67,23 +69,54 @@ Future<void> init(Args args) async {
         await windowManager.show();
         await windowManager.focus();
       }
+      // Apply menubar mode (hide dock icon) on macOS - must be after show()
```
I have tested that the dock icon can be hidden through the skipTaskbar option in windowOptions above, so there is no need for this code here.

monkeyWie commented on ui/flutter/lib/main.dart:
```diff
@@ -67,23 +69,54 @@ Future<void> init(Args args) async {
       center: true,
-      skipTaskbar: false,
+      skipTaskbar: runAsMenubarApp,
+      title: 'Gopeed',
```
I don't want this title, otherwise it's ugly~

awe123343 replied:
nvm, below is the right fix
```swift
    super.applicationWillFinishLaunching(notification)
```

awe123343 commented:
> I’ve looked into the current situation, and here are the adjustments you need to make:
> 
> 1. Please downgrade to Flutter 3.24 for development.
> 2. Please revert the native Swift changes. Instead, implement the logic to hide the Dock icon in the onWindowFocus callback within app_controller.dart. This should fix the issue where the Dock icon reappears when the app is reopened.
> 3. Do not commit files under macos/Runner.xcodeproj.
> 4. This allows to remove the dependency on shared_preferences.
> 
> After making these changes, if the CI passes, we can merge this PR. Thanks!

Thanks for looking into it! I have made changes accordingly.
It looks like CI need maintainer's approval to run?

monkeyWie commented:
Now only the Linux CI has failed, according to the hotkey_manager documentation, the keybinder-3.0 dep needs to be installed, also need modify `.github/workflows/test.yml` and `.github/workflows/build.yml` to fix it.

awe123343 commented:
> Now only the Linux CI has failed, according to the hotkey_manager documentation, the keybinder-3.0 dep needs to be installed, also need modify `.github/workflows/test.yml` and `.github/workflows/build.yml` to fix it.

Thx, updated, also updated `ui/flutter/linux/packaging/deb/make_config.yaml
